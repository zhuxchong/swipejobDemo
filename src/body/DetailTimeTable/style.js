import styled from "styled-components";
export const DetailTimeTableWrapper = styled.div`
  width: 100%;
  background-color: transparent;
  padding: 1% 0%;
  display: flex;
  flex-direction: column;
  flex: 5;
  p {
    text-align: center;
    line-height: 25px;
    margin: 0%;
    font-size: 18px;
  }
  h1 {
    text-align: center;
    margin-top: 0px;
    margin-bottom: 10px;
    font-size: 20;
  }
  @media (max-width: 768px) {
    padding: 1% 0%;
    h1 {
      font-size: 15px;
      margin-bottom: 0px;
    }
    p {
      color: gray;
      text-indent: 10px;
      font-size: 12px;
      line-height: 15px;
    }
  }
  /*for iphone x/6/7/8/plus*/
  @media (max-height: 823px) and (min-height: 667px) {
    p {
      color: gray;
      text-indent: 10px;
      font-size: 18px;
      line-height: 20px;
      margin-top: 5px;
    }
  }
  /*for ipad*/
  @media (max-height: 1025px) and (min-height: 812px) {
    h1 {
      font-size: 20px;
      margin-bottom: 0px;
    }
    p {
      color: gray;
      text-indent: 10px;
      font-size: 20px;
      line-height: 25px;
      margin-top: 5px;
    }
  }
  /*for ipad pro*/
  @media (max-height: 1366px) and (min-height: 1026px) {
    h1 {
      font-size: 30px;
      margin-bottom: 0px;
    }
    p {
      text-indent: 10px;
      font-size: 30px;
      line-height: 40px;
      margin-top: 10px;
    }
  }
`;
