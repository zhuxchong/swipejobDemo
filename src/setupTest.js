import { configure } from "enzyme";
import Ada from "enzyme-adapter-react-16";
//set up
configure({ adapter: new Ada() });
