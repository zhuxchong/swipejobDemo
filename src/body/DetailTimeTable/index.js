import React from "react";
import { DetailTimeTableWrapper } from "./style";
import { Button } from "antd";

class DetailTimeTable extends React.Component {
  render() {
    const { page, totalPage } = this.props;
    return (
      <DetailTimeTableWrapper>
        {/* {console.log(this.props.match.params.id)} */}
        <h1>If you take this job you are agreeing to work ALL DAYS</h1>
        <Button
          type="primary"
          size="small"
          style={{ width: "60px", textAlign: "center", margin: "10px auto" }}
          className="antButton"
          onClick={() => {
            this.handleChagePage(page, totalPage);
          }}
        >
          {`${page}/${totalPage}`}
        </Button>
        {this.getListArea().map(i => {
          return i;
        })}
      </DetailTimeTableWrapper>
    );
  }
  getListArea = () => {
    const newList = [];
    //console.log(this.props.workHour);
    if (this.props.workHour.length) {
      for (let i = (this.props.page - 1) * 4; i < this.props.page * 4; i++) {
        if (this.props.workHour[i]) {
          newList.push(
            <p key={this.props.workHour[i]} className="detailTime">{`${
              this.props.workHour[i][0]
            }, ${this.props.workHour[i][1]} ${this.props.workHour[i][2]} ${
              this.props.workHour[i][3]
            }`}</p>
          );
        }
      }
    }
    return newList;
  };
  handleChagePage = (page, totalPage) => {
    this.props.handChagePage(page, totalPage);
  };
  //<div>123{console.log(props)}</div>
}
export default DetailTimeTable;
