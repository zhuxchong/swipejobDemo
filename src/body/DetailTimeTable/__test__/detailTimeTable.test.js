import React from "react";
import { shallow } from "enzyme";
import DetailTimeTable from "../index";
require("../../../setupTest");
const props = {
  //   title: "Construction and Releated Workers - General",
  //   companyName: "C.D. Barnes and Associates",
  //   wagePerHour: "$8.38/hour",
  //   workPeriod: "Mon, Sep 05 - Fri, Sep 16",
  handleChagePage: jest.fn(),
  handChagePage: jest.fn(),
  page: 1,
  totalPage: 3,
  workHour: [
    ["Mon", "Sep", "05", "7 AM"],
    ["Tue", "Sep", "06", "7 AM"],
    ["Tue", "Sep", "06", "7 AM"],
    ["Tue", "Sep", "06", "7 AM"],
    ["Tue", "Sep", "06", "7 AM"],
    ["Tue", "Sep", "06", "7 AM"],
    ["Tue", "Sep", "06", "7 AM"],
    ["Tue", "Sep", "06", "7 AM"],
    ["Tue", "Sep", "06", "7 AM"],
    ["Tue", "Sep", "06", "7 AM"]
  ]
};
let wrapper;
describe("DetailTimeTable component", () => {
  beforeEach(() => {
    wrapper = shallow(<DetailTimeTable {...props} />);
  });
  it("should render the component to match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should trigger the correct function callbacks", () => {
    const btn = wrapper.find(".antButton").simulate("click");
    expect(props.handChagePage).toHaveBeenCalled();
  });
  it("should render correct items and time for each", () => {
    const pList = wrapper.find(".detailTime");
    expect(pList.length).toEqual(4);
  });
});
