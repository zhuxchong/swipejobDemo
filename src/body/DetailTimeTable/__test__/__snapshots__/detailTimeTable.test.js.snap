// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailTimeTable component should render the component to match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DetailTimeTable
    handChagePage={[MockFunction]}
    handleChagePage={[MockFunction]}
    page={1}
    totalPage={3}
    workHour={
      Array [
        Array [
          "Mon",
          "Sep",
          "05",
          "7 AM",
        ],
        Array [
          "Tue",
          "Sep",
          "06",
          "7 AM",
        ],
        Array [
          "Tue",
          "Sep",
          "06",
          "7 AM",
        ],
        Array [
          "Tue",
          "Sep",
          "06",
          "7 AM",
        ],
        Array [
          "Tue",
          "Sep",
          "06",
          "7 AM",
        ],
        Array [
          "Tue",
          "Sep",
          "06",
          "7 AM",
        ],
        Array [
          "Tue",
          "Sep",
          "06",
          "7 AM",
        ],
        Array [
          "Tue",
          "Sep",
          "06",
          "7 AM",
        ],
        Array [
          "Tue",
          "Sep",
          "06",
          "7 AM",
        ],
        Array [
          "Tue",
          "Sep",
          "06",
          "7 AM",
        ],
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h1>
          If you take this job you are agreeing to work ALL DAYS
        </h1>,
        <Button
          block={false}
          className="antButton"
          ghost={false}
          loading={false}
          onClick={[Function]}
          size="small"
          style={
            Object {
              "margin": "10px auto",
              "textAlign": "center",
              "width": "60px",
            }
          }
          type="primary"
        >
          1/3
        </Button>,
        Array [
          <p
            className="detailTime"
          >
            Mon, Sep 05 7 AM
          </p>,
          <p
            className="detailTime"
          >
            Tue, Sep 06 7 AM
          </p>,
          <p
            className="detailTime"
          >
            Tue, Sep 06 7 AM
          </p>,
          <p
            className="detailTime"
          >
            Tue, Sep 06 7 AM
          </p>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "If you take this job you are agreeing to work ALL DAYS",
        },
        "ref": null,
        "rendered": "If you take this job you are agreeing to work ALL DAYS",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": false,
          "children": "1/3",
          "className": "antButton",
          "ghost": false,
          "loading": false,
          "onClick": [Function],
          "size": "small",
          "style": Object {
            "margin": "10px auto",
            "textAlign": "center",
            "width": "60px",
          },
          "type": "primary",
        },
        "ref": null,
        "rendered": "1/3",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Mon,Sep,05,7 AM",
        "nodeType": "host",
        "props": Object {
          "children": "Mon, Sep 05 7 AM",
          "className": "detailTime",
        },
        "ref": null,
        "rendered": "Mon, Sep 05 7 AM",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "Tue,Sep,06,7 AM",
        "nodeType": "host",
        "props": Object {
          "children": "Tue, Sep 06 7 AM",
          "className": "detailTime",
        },
        "ref": null,
        "rendered": "Tue, Sep 06 7 AM",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "Tue,Sep,06,7 AM",
        "nodeType": "host",
        "props": Object {
          "children": "Tue, Sep 06 7 AM",
          "className": "detailTime",
        },
        "ref": null,
        "rendered": "Tue, Sep 06 7 AM",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "Tue,Sep,06,7 AM",
        "nodeType": "host",
        "props": Object {
          "children": "Tue, Sep 06 7 AM",
          "className": "detailTime",
        },
        "ref": null,
        "rendered": "Tue, Sep 06 7 AM",
        "type": "p",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  width: 100%;
  background-color: transparent;
  padding: 1% 0%;
  display: flex;
  flex-direction: column;
  flex: 5;
  p {
    text-align: center;
    line-height: 25px;
    margin: 0%;
    font-size: 18px;
  }
  h1 {
    text-align: center;
    margin-top: 0px;
    margin-bottom: 10px;
    font-size: 20;
  }
  @media (max-width: 768px) {
    padding: 1% 0%;
    h1 {
      font-size: 15px;
      margin-bottom: 0px;
    }
    p {
      color: gray;
      text-indent: 10px;
      font-size: 12px;
      line-height: 15px;
    }
  }
  /*for iphone x/6/7/8/plus*/
  @media (max-height: 823px) and (min-height: 667px) {
    p {
      color: gray;
      text-indent: 10px;
      font-size: 18px;
      line-height: 20px;
      margin-top: 5px;
    }
  }
  /*for ipad*/
  @media (max-height: 1025px) and (min-height: 812px) {
    h1 {
      font-size: 20px;
      margin-bottom: 0px;
    }
    p {
      color: gray;
      text-indent: 10px;
      font-size: 20px;
      line-height: 25px;
      margin-top: 5px;
    }
  }
  /*for ipad pro*/
  @media (max-height: 1366px) and (min-height: 1026px) {
    h1 {
      font-size: 30px;
      margin-bottom: 0px;
    }
    p {
      text-indent: 10px;
      font-size: 30px;
      line-height: 40px;
      margin-top: 10px;
    }
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h1>
            If you take this job you are agreeing to work ALL DAYS
          </h1>,
          <Button
            block={false}
            className="antButton"
            ghost={false}
            loading={false}
            onClick={[Function]}
            size="small"
            style={
              Object {
                "margin": "10px auto",
                "textAlign": "center",
                "width": "60px",
              }
            }
            type="primary"
          >
            1/3
          </Button>,
          Array [
            <p
              className="detailTime"
            >
              Mon, Sep 05 7 AM
            </p>,
            <p
              className="detailTime"
            >
              Tue, Sep 06 7 AM
            </p>,
            <p
              className="detailTime"
            >
              Tue, Sep 06 7 AM
            </p>,
            <p
              className="detailTime"
            >
              Tue, Sep 06 7 AM
            </p>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "If you take this job you are agreeing to work ALL DAYS",
          },
          "ref": null,
          "rendered": "If you take this job you are agreeing to work ALL DAYS",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "children": "1/3",
            "className": "antButton",
            "ghost": false,
            "loading": false,
            "onClick": [Function],
            "size": "small",
            "style": Object {
              "margin": "10px auto",
              "textAlign": "center",
              "width": "60px",
            },
            "type": "primary",
          },
          "ref": null,
          "rendered": "1/3",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Mon,Sep,05,7 AM",
          "nodeType": "host",
          "props": Object {
            "children": "Mon, Sep 05 7 AM",
            "className": "detailTime",
          },
          "ref": null,
          "rendered": "Mon, Sep 05 7 AM",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "Tue,Sep,06,7 AM",
          "nodeType": "host",
          "props": Object {
            "children": "Tue, Sep 06 7 AM",
            "className": "detailTime",
          },
          "ref": null,
          "rendered": "Tue, Sep 06 7 AM",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "Tue,Sep,06,7 AM",
          "nodeType": "host",
          "props": Object {
            "children": "Tue, Sep 06 7 AM",
            "className": "detailTime",
          },
          "ref": null,
          "rendered": "Tue, Sep 06 7 AM",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "Tue,Sep,06,7 AM",
          "nodeType": "host",
          "props": Object {
            "children": "Tue, Sep 06 7 AM",
            "className": "detailTime",
          },
          "ref": null,
          "rendered": "Tue, Sep 06 7 AM",
          "type": "p",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  width: 100%;
  background-color: transparent;
  padding: 1% 0%;
  display: flex;
  flex-direction: column;
  flex: 5;
  p {
    text-align: center;
    line-height: 25px;
    margin: 0%;
    font-size: 18px;
  }
  h1 {
    text-align: center;
    margin-top: 0px;
    margin-bottom: 10px;
    font-size: 20;
  }
  @media (max-width: 768px) {
    padding: 1% 0%;
    h1 {
      font-size: 15px;
      margin-bottom: 0px;
    }
    p {
      color: gray;
      text-indent: 10px;
      font-size: 12px;
      line-height: 15px;
    }
  }
  /*for iphone x/6/7/8/plus*/
  @media (max-height: 823px) and (min-height: 667px) {
    p {
      color: gray;
      text-indent: 10px;
      font-size: 18px;
      line-height: 20px;
      margin-top: 5px;
    }
  }
  /*for ipad*/
  @media (max-height: 1025px) and (min-height: 812px) {
    h1 {
      font-size: 20px;
      margin-bottom: 0px;
    }
    p {
      color: gray;
      text-indent: 10px;
      font-size: 20px;
      line-height: 25px;
      margin-top: 5px;
    }
  }
  /*for ipad pro*/
  @media (max-height: 1366px) and (min-height: 1026px) {
    h1 {
      font-size: 30px;
      margin-bottom: 0px;
    }
    p {
      text-indent: 10px;
      font-size: 30px;
      line-height: 40px;
      margin-top: 10px;
    }
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
